### ISSUE TITLE
Problems during training

### ISSUE THREAD

**Issue by Jiafei127** [ZhengPeng7/motion_magnification_learning-based#11]
Hello, is it convenient to ask a question?
Why I haven't made a stable state in the loss region after 10 days of training with 12 batch size? Did you fit the loss during the training?

**Comment by Jiafei127** [comment]
> Just try to use the weights you've already obtained to make an inference and see whether it works well.
> If you use the given dataset in this paper, there is no validation set.

Thank you! But why is that? It sounds quite incomprehensible.
Is it because of the specificity of this task?
Does the loss setting cause it to only pull in the distance from the true value and not fit it?

**Comment by ZhengPeng7** [comment]
What do you exactly mean by the  'stable state'? If you only see the training loss, of couse it will decrease even after a very long time. I think the settings about loss and training in this code doesn't have problem as others and I tried.

**Comment by Jiafei127** [comment]
I'm sorry I didn't explain it clearly, but I don't quite understand how to determine the "completion" of the training process. Because I noticed during training that after a long time, the loss was still dropping.

**Comment by ZhengPeng7** [comment]
Yeah, I understood what you said nad that's what I meant. In my previous training, I found the weights obtained at 12-th epoch is okay. You can use that to have a try for inference.

**Comment by Jiafei127** [comment]
Okey... Makes me wonder if the results don't get better as the training time increases, so that there's no way to know how to get one of the best results.
But thank you very much! Also, line 55 of your program (test_video.py) is just a type of test, called "dynamic mode" in the original. If batch_A is the first frame, then it is "static mode".

**Comment by ZhengPeng7** [comment]
The VMM is currently only evaluated by visualization instead of some metrics, so we cannot exactly know if the model is still getting improved in training.
Thanks, I know, I made these two modes in my previous experiments (but maybe not in this repo). I think the mode here is the dynamic mode (magnify by the two neighbouring frames instead of the first and the current frame).

**Comment by Jiafei127** [comment]
Okey... Well... It amazes me. Wouldn't the results get better as the training time increases? If so there would be no way to know when to get a best result.
But thank you very much! Also, line 55 of your program (test_video.py) is just a type of test, called "dynamic mode" in the original. If batch_A is the first frame, then it is "static mode".

**Comment by Jiafei127** [comment]
Ok, I appreciate the discussion with you.

**Comment by ZhengPeng7** [comment]
> Okey... Well... It amazes me. Wouldn't the results get better as the training time increases? If so there would be no way to know when to get a best result. But thank you very much! Also, line 55 of your program (test_video.py) is just a type of test, called "dynamic mode" in the original. If batch_A is the first frame, then it is "static mode".

The model may already converge. So too much training tends to cause overfitting on the training set.

**Comment by ZhengPeng7** [comment]
You are welcome :)

**Comment by Jiafei127** [comment]
Zooming in on the differences between adjacent frames is mainly for particularly violent movements, and is more similar to the previous method of using acceleration zoom. This is how I see it.

**Comment by ZhengPeng7** [comment]
Yeah, there is difference between these two modes. I think it's not hard to imagine by some casual drawings of moving points.
